src/App.tsx

```diff
@@ -14,11 +14,132 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const [isShown, setIsShown] = useState(false);
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [length, setLength] = useState('1');
+
+  const reverse = () => {
+    setGoods([...goods].reverse().filter(good => good.length >= +length));
+  };
+
+  const sort = () => {
+    setGoods([...goods]
+      .sort((a, b) => a.localeCompare(b))
+      .filter(good => good.length >= +length));
+  };
+
+  const reset = () => {
+    setGoods(goodsFromServer);
+    setLength('1');
+  };
+
+  const sortByLength = () => {
+    setGoods([...goods].sort((a, b) => b.length - a.length));
```

Comment:
you have 2 functions with almost similar logic of sort, combine them

Reply:
What do you mean?

Reply:
u can create a function that gonna accepts the type of sort and sort it in that case, store this type in state, and change it on click. In function it is better to use switch

Reply:
I don't want to do it